networking/change-block-size.md

```diff
@@ -0,0 +1,209 @@
+
+### How to
+
+>**Important!** Follow the steps in order to minimize pod connectivity disruption. Pods may lose connectivity when they are redeployed, and may lose external connectivity while in the temporary pool. Also, when pods are deleted, applications may be temporarily unavailable (depending on the type of application). Plan your changes accordingly. 
+{: .alert .alert-danger}
+
+The high-level steps to follow are:
+
+1. [Create a temporary IP pool](#create-a-temporary-ip-pool)    
+  **Note**: The temporary IP pool must not overlap with the existing one.
+1. [Disable the existing IP pool](#disable-the-existing-ip-pool)    
+  **Note**: When you disable an IP pool, only new IP address allocations are prevented; networking of existing pods are not affected.
+1. [Delete pods from the existing IP pool](#delete-pods-from-the-existing-ip-pool)    
+   This includes any new pods that may have been created with the existing IP pool prior to disabling the pool. Verify that new pods get an address from the temporary IP pool.
+1. [Delete the existing IP pool](#delete-the-existing-ip-pool)
+1. [Create a new IP pool with the desired block size](#create-a-new-ip-pool-with-the-desired-block-size)
+1. [Disable the temporary IP pool](#disable-the-temporary-ip-pool)
+1. [Delete pods from the temporary IP pool](#delete-pods-from-the-temporary-ip-pool)
+1. [Delete the temporary IP pool](#delete-the-temporary-ip-pool)
+
+### Tutorial
+
+In the following steps, our Kubernetes cluster has a default CIDR block size of /26. We want to shrink the block size to /28 to use the pool more efficiently. 
+
+#### Create a temporary IP pool
+
+We add a new IPPool with the CIDR range, 10.0.0.0/16.
+
+Create a temporary-pool.yaml.
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: temporary-pool
+spec:
+  cidr: 10.0.0.0/16
+  ipipMode: Always
+  natOutgoing: true
+</pre>
+
+Apply the changes.
+
+```
+calicoctl apply -f temporary-pool.yaml
+```
+
+Let’s verify the temporary IP pool.
+
+```
+calicoctl get ippool -o wide
+```
+
+<pre>
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     false
+temporary-pool        10.0.0.0/16      true   Always     false
+</pre>
+
+#### Disable the existing IP pool
+
+List the existing IP pool definition.
+
+```
+calicoctl patch ippool default-ipv4-ippool -p '{"spec": {"disabled": “true”}}'
+```
+
+Verify the changes.
+
+```
+calicoctl get ippool -o wide
+```
+
+<pre>
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     true
+temporary-pool        10.0.0.0/16      true   Always     false
+</pre>
+
+#### Delete pods from the existing IP pool
+
+In our example, **coredns** is our only pod; for multiple pods you would trigger a deletion for all pods in the cluster.
+
+```
+kubectl delete pod -n kube-system coredns-6f4fd4bdf-8q7zp
```

Comment:
I think a "nicer" command for this would be something like this:

```
kubectl rollout restart -n kube-system coredns
```

This doesn't delete the pod directly, but tells Kubernetes to do a rolling update of the deployment managing the pod. So, for "real" applications this is going to handle things like multiple replicas, etc.

Reply:
But probably fine to leave as-is for this guide, I guess.

Reply:
I think having both are good so added "OR"